controllers/performanceprofile_controller.go

```diff
@@ -540,3 +542,34 @@ func hasMatchingLabels(performanceprofile *performancev2.PerformanceProfile, mcp
 	}
 	return true
 }
+
+// CompList a map that contains Kind string as key and component's instances as value
+type CompList map[string]interface{}
```

Comment:
Well, I'd use a plain old struct like
```
type ManifestResultSet struct {
       MachineConfig *machineconfigv1.MachineConfig
       KubeletConfig *machineconfigv1.KubeletConfig
       Tuned         *tunedv1.Tuned
       RuntimeClass  *nodev1beta1.RuntimeClass
}
```
Not like we add new outputs very day (or even every major...)
This saves type assertions and looks a bit neater.

Reply:
But then we won't be able to iterate the struct, when most of the action are common and can be done using loops.

Reply:
see below about `ToObjects`. Lacking generics, lacking better ideas (which admittledly I don't have now) we prefer quasi-duplicate, type-specific code to scattered type assertions.

Reply:
yet you'll need a way to iterate the struct inside the ```ToObject``` in order to create this ```[]v1.Object``` but I see your point. type assertion might be confusing and less readable

Reply:
In `ToObject` we should convert each field explicitely (no generic iteration). The idea is to keep at least all the conversion in a single place and let not them spread around in the codebase.